test/integration/test_media.py

```diff
@@ -0,0 +1,204 @@
+    conservator.media.wait_for_processing(media_id, check_frequency_seconds=1)
+    image = conservator.get_media_instance_from_id(media_id)
+    assert image is not None
+    assert isinstance(image, Image)
+
+    frame_0 = image.get_frame_by_index(0)
+    assert frame_0 is not None
+    assert frame_0.video_id == image.id
+    assert frame_0.video_name == image.name
+    assert frame_0.height == 375
+    assert frame_0.width == 500
+
+    with pytest.raises(IndexError):
+        image.get_frame_by_index(-1)
+
+    with pytest.raises(IndexError):
+        image.get_frame_by_index(1)
+
+
+def test_image_get_frame(conservator, test_data):
+    path = test_data / "jpg" / "cat_0.jpg"
+    media_id = conservator.media.upload(path)
+    conservator.media.wait_for_processing(media_id, check_frequency_seconds=1)
+    image = conservator.get_media_instance_from_id(media_id)
+    assert image is not None
+    assert isinstance(image, Image)
+
+    frame = image.get_frame()
+    assert frame is not None
+    assert frame.video_id == image.id
+    assert frame.video_name == image.name
+    assert frame.height == image.height
+    assert frame.width == image.width
+
+
+def test_get_all_frames_paginated(conservator, test_data):
+    path = test_data / "mp4" / "adas_thermal.mp4"
+    media_id = conservator.media.upload(path)
+    conservator.media.wait_for_processing(media_id, check_frequency_seconds=1)
+    video = conservator.get_media_instance_from_id(media_id)
+    assert video is not None
+    assert isinstance(video, Video)
+
+    paginated_frames = video.get_all_frames_paginated()
+    frames = list(paginated_frames)
+    assert len(frames) == video.frames_count
+    for i, frame in enumerate(frames):
+        assert frame.frame_index == i
+        assert frame.video_id == video.id
+        assert frame.video_name == video.name
+        assert frame.width == video.width
+        assert frame.height == video.height
+
+
+def test_compare_media(conservator, test_data):
+    path = test_data / "jpg" / "cat_0.jpg"
+    media_id = conservator.media.upload(path)
+    conservator.media.wait_for_processing(media_id, check_frequency_seconds=1)
+    image = conservator.get_media_instance_from_id(media_id)
+
+    assert image.compare(path) == MediaCompare.MATCH
+    assert image.compare(path).ok()
+
+    wrong_path = test_data / "jpg" / "cat_1.jpg"
+    assert image.compare(wrong_path) == MediaCompare.MISMATCH
+    assert not image.compare(wrong_path).ok()
+
+
+def test_get_annotations_empty(conservator, test_data):
+    path = test_data / "jpg" / "cat_0.jpg"
+    media_id = conservator.media.upload(path)
+    conservator.media.wait_for_processing(media_id, check_frequency_seconds=1)
+    image = conservator.get_media_instance_from_id(media_id)
+
+    annotations = image.get_annotations()
+    assert len(annotations) == 0
+
+
+def test_get_annotations(conservator, test_data):
+    # TODO: Test once we have a way to add annotations
+    pass
+
```

Comment:
We could also check in a pre-populated metadata file and upload it to the media file, then verify the annotations are properly retrieved... but maybe that test would fit well into a separate yet-to-come set of tests for metadata.

Reply:
Ah good idea--that's probably the best way to test annotations ATM. I'll use it here once I've got some test metadata in the repo